docs/mentoring.md

```diff
@@ -1,60 +0,0 @@
-# Mentoring
```

Comment:
do we want to rewrite this page to instead encourage questions on issues and PRs? I can rewrite it if so